integration/tests/epochs/base_suite.go

```diff
@@ -150,8 +159,53 @@ func (s *BaseSuite) AwaitEpochPhase(ctx context.Context, expectedEpoch uint64, e
 }
 
 // GetContainersByRole returns all containers from the network for the specified role, making sure the containers are not ghost nodes.
+// Since go maps have random iteration order the list of containers returned will be in random order.
 func (s *BaseSuite) GetContainersByRole(role flow.Role) []*testnet.Container {
-	nodes := s.net.ContainersByRole(role, false)
+	nodes := s.Net.ContainersByRole(role, false)
 	require.True(s.T(), len(nodes) > 0)
 	return nodes
 }
+
+// AwaitFinalizedView polls until it observes that the latest finalized block has a view
+// greater than or equal to the input view. This is used to wait until when an epoch
+// transition must have happened.
+func (s *BaseSuite) AwaitFinalizedView(ctx context.Context, view uint64, waitFor, tick time.Duration) {
+	require.Eventually(s.T(), func() bool {
+		finalized := s.GetLatestFinalizedHeader(ctx)
+		return finalized.View >= view
+	}, waitFor, tick)
+}
+
+// GetLatestFinalizedHeader retrieves the latest finalized block, as reported in LatestSnapshot.
+func (s *BaseSuite) GetLatestFinalizedHeader(ctx context.Context) *flow.Header {
+	snapshot := s.GetLatestProtocolSnapshot(ctx)
+	finalized, err := snapshot.Head()
+	require.NoError(s.T(), err)
+	return finalized
+}
+
+// AssertInEpoch requires actual epoch counter is equal to counter provided.
```

Comment:
not sure that the phrase "as of the latest finalized block" is grammatically correct. If it is not, I hope you understand what I want to say and correct my suggestion accordingly. 

```suggestion
// AssertInEpoch requires that the current epoch's counter (as of the latest finalized block) is equal to the counter value provided.
```